src/nvim/eval.c

```diff
@@ -15140,8 +15140,11 @@ static void f_sockconnect(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     return;
   }
 
+  snprintf((char *)IObuff, sizeof(IObuff), "%s:%" PRIdLINENR,
```

Comment:
Channel module needs no "digging around", the function can be in eval.c for the moment, and can be generalized later to handle lua tracebacks and remote calls. I think it is an occasion where dynamic scoping makes sense, rather than ending up passing this information to every function that ever does logging, as logging we might want to do in any function.